CRM/Contact/BAO/Query.php

```diff
@@ -4200,7 +4200,10 @@ public function relationship(&$values) {
     $relationshipTempTable = NULL;
     if (self::$_relType == 'reciprocal') {
       $where = [];
-      self::$_relationshipTempTable = $relationshipTempTable = CRM_Core_DAO::createTempTableName('civicrm_rel');
+      self::$_relationshipTempTable = $relationshipTempTable = CRM_Utils_SQL_TempTable::build()
+        ->setCategory('civicrm_rel')
```

Comment:
__BG__: `createTempTableName($prefix)` and `tempTable->category` are not interchangeable. The former yields a table name like `civicrm_rel_temp_1234` while the latter would look like `civicrm_tmp_e_civicrm_rel_1234`. The `category` has tighter validation.

__Comment__: When I manually run ` TempTable::build()` like this in `cv`, it throws an exception on the `_`. Assuming that this change is safe (e.g. there's no other code that relies on the `/^civicrm_rel_/` prefix), then you could just do `rel` or drop the category completely.